packages/web-config-server/src/apiV1/dataBuilders/helpers/calculateOperationForAnalytics.js

```diff
@@ -79,13 +81,47 @@ const combineBinaryIndicatorsToString = (analytics, config) => {
   return stringArray.length === 0 ? 'None' : stringArray.join(', ');
 };
 
+const combineTextIndicators = (analytics, config) => {
+  const { dataElements } = config;
+  const filteredAnalytics = analytics.filter(({ dataElement: de }) => dataElements.includes(de));
+  const stringArray = [];
+  filteredAnalytics.forEach(({ dataElement, value }) => {
+    if (value) {
+      stringArray.push(value);
+    }
+  });
+  return stringArray.length === 0 ? 'None' : stringArray.join(', ');
+};
+
+const getMetaDataFromOrgUnit = async (_, config) => {
+  const { orgUnitCode, ancestorType, field, jsonPath, hierarchyId } = config;
+  const baseEntity = await Entity.findOne({ code: orgUnitCode });
+  if (!baseEntity) return 'Entity not found';
+  const entity = ancestorType
+    ? await baseEntity.getAncestorOfType(ancestorType, hierarchyId)
+    : baseEntity;
+
+  switch (field) {
+    case 'type':
```

Comment:
Should we change this to another name since it can get mixed up with the actual type of the entity?

Reply:
Good idea 😬, didn't think about that.